--- a/include/linux/mtd/partitions.h
+++ b/include/linux/mtd/partitions.h
@@ -83,6 +83,11 @@
 int mtd_add_partition(struct mtd_info *master, char *name,
 		      long long offset, long long length);
 int mtd_del_partition(struct mtd_info *master, int partno);
+
+struct mtd_info *mtdpart_get_master(const struct mtd_info *mtd);
+uint64_t mtdpart_get_offset(const struct mtd_info *mtd);
+uint64_t mtd_get_device_size(const struct mtd_info *mtd);
+
 extern void __weak arch_split_mtd_part(struct mtd_info *master,
 		      const char *name, int offset, int size);
 
