--- a/drivers/mtd/devices/hisfc350/hisfc350_spi_ids.c	2019-07-11 18:16:34.000000000 +0300
+++ b/drivers/mtd/devices/hisfc350/hisfc350_spi_ids.c	2019-07-11 22:09:23.933611095 +0300
@@ -1891,6 +1891,32 @@
 		&spi_driver_en25q64,
 	},
 
+/*need to add dual rw*/
+	{
+		"EN25QH64", {0x1c, 0x70, 0x17}, 3, (_64K * 128),  _64K, 3,
+		{
+			&READ_STD(0, INFINITE, 50),
+			&READ_FAST(1, INFINITE, 100),
+			&READ_DUAL(1, INFINITE, 80),
+			&READ_DUAL_ADDR(1, INFINITE, 80),
+#ifndef CONFIG_CLOSE_SPI_8PIN_4IO
+			&READ_QUAD_ADDR(3, INFINITE, 50),
+#endif
+			0
+		},
+
+		{
+			&WRITE_STD(0, 256, 80),
+			0
+		},
+
+		{
+			&ERASE_SECTOR_64K(0, _64K, 80/*104*/),
+			0
+		},
+		&spi_driver_generic,
+	},
+
 	{
 		"EN25Q128", {0x1c, 0x30, 0x18}, 3, (_64K * 256),  _64K, 3,
 		{
