--- a/drivers/mtd/Kconfig
+++ b/drivers/mtd/Kconfig
@@ -37,6 +37,17 @@ config MTD_UIMAGE_SPLIT
 	depends on MTD_SPLIT_FIRMWARE
 	default y
 
+comment "Rootfs partition parsers"
+
+config MTD_SPLIT_SQUASHFS_ROOT
+	bool "Squashfs based root partition parser"
+	select MTD_SPLIT
+	default n
+	help
+	  This provides a parsing function which allows to detect the
+	  offset and size of the unused portion of a rootfs partition
+	  containing a squashfs.
+
 config MTD_SPLIT
 	def_bool n
 	help 
