diff -Naur a/drivers/mmc/host/himciv100/himci.c b/drivers/mmc/host/himciv100/himci.c
--- a/drivers/mmc/host/himciv100/himci.c	2019-08-04 16:37:10.740926869 +0200
+++ b/drivers/mmc/host/himciv100/himci.c	2019-08-01 23:29:43.159039282 +0200
@@ -92,6 +92,9 @@
 	local_irq_save(flags);
 
 	hi_mci_ctr_reset();
+	udelay(50);
+
+	hi_mci_ctr_undo_reset();
 
 	reg_value = himci_readl(host->base + MCI_BMOD);
 	reg_value |= BMOD_SWR;
@@ -109,17 +112,6 @@
 	local_irq_restore(flags);
 }
 
-static void hi_mci_sys_undo_reset(struct himci_host *host)
-{
-	unsigned long flags;
-
-	himci_trace(2, "undo reset");
-
-	local_irq_save(flags);
-	hi_mci_ctr_undo_reset();
-	local_irq_restore(flags);
-}
-
 static void hi_mci_ctrl_power(struct himci_host *host, unsigned int flag)
 {
 	himci_trace(2, "begin");
@@ -264,8 +256,6 @@
 	hi_mci_ctrl_power(host, POWER_ON);
 	udelay(200);
 
-	hi_mci_sys_undo_reset(host);
-
 	/* clear MMC host intr */
 	himci_writel(ALL_INT_CLR, host->base + MCI_RINTSTS);
 
